Fix too many requests error

Burge337 opened the pull request:
Still has some weirdness in that using the site in Chrome is significantly slower than Firefox, but even then the delay isn't super terrible. 
Double clicking when performing a non-destructive action will still duplicate that action. 
Testing should be complete for the new methods, although the errorIsThrownWhenDecodingFails test in the TokenVerifierSpec file is commented out because it only succeeds in the JUnit tests. (something about gradle not knowing what context.status() is)

Closes #19

Burge337 committed:
Remove extra before server calls for verification

I feel like the before calls under get a single note should cover most of the other methods because they all use the api/notes/:id routing, and while the edit note routing is different, it calls getNoteByID and hits the other routing anyway. I'm committing this as a single file change so it'll be easier to change this if I messed something up.

#19

Burge337 committed:
Add sub string to owner fields in database

I'm doing so with the intention of using the subject to get whatever user info we want from our database instead of always grabbing stuff from auth0. (as per the discussion in the slack thread)

Burge337 committed:
Fix owner component

It broke when the sub field was added to owners. I just added that field to the new owner creation method.

#19

Burge337 committed:
Fix failing client tests

Forgot to add the new owner sub field to the mock service

#19

Burge337 committed:
Create method to get subject from the JWT token in the context

There are also tests for this, but one of them only works in the spec file itself, and breaks when using gradlew test. More specifically, it gives a java.lang.NoSuchMethodError at TokenVerifierSpec.java:338, which seems to take issue with ctx.status().

#19

Burge337 committed:
Delete old unused methods and comment out tokenverifier test for now

The test works with run test, but fails with gradle. I've commented this out for now just so I can see of github actions takes issue with any of the other tests.

#19

Burge337 committed:
Make various methods use the subject to request info from the DB

This has fixed the too many requests error as far as I can tell.

Some important things to note however:
1) The site runs slowly on chrome. (It runs really well on firefox for some reason though)
2) The new methods still need to have tests written for them. (next commit will focus on this)
3) The GetOwners() test was failing in the spec, but I don't think we ever use this method anyhow, so I've commented it out and plan on deleting it.

#19

Burge337 committed:
Remove old print lines

#19

Burge337 committed:
Add JUnit tests for GetOwnerIDBySubject

I also got rid of getOwnerBySubject because it never got called.

#19

Burge337 committed:
Fix and add to owner.service.spec tests

I don't know how to test getSignUrl, so we still need that test done, but other than that we should have owner.service covered.

#19

helloworld12321 commented:
Looks like GitHub Actions is flaking out again :-(

I ran the tests locally, though, and everything seems A-okay! :relaxed:

helloworld12321 committed:
Move around the 'before' clauses

With Ben Goldstein, we found out that:

- Sub-paths do not share the before clause of their parent: that is to say, if `/aaa` has a before clause, it will not run for `/aaa/bbb`.

- That being said, if two endpoints have different methods, they will share a before clause. For example, if there is a before clause on `GET /aaa`, it will also run for `POST /aaa` and `DELETE /aaa`.

This commit moves around the before clauses so that only one of them runs on any given endpoint :)

helloworld12321 committed:
Remove the specialized 404 messages on certain owner endpoints

I talked to Ben Burgess, and he said these were mostly for debugging purpose—seems like we can remove them in production! :)

helloworld12321 commented:
For the record, I'm not seeing any of the slowness you mentioned on Chrome. It seems to be working just fine on all browsers :)
